### PR TITLE
readmanga.today implementation

### PR DESCRIPTION
**Pull request by Vesnyx**
This is incomplete at the moment, but I'll be fixing the rest of it in the near future. Only reason I'm requesting this is because I have no way of testing the PHP for debugging purposes. I'm not even the slightest bit experienced in it for that matter. I've tried my best to write the model, but I'm not quite sure if it works. 

EDIT: 😂 gosh darn typos

### CODE REVIEW

File: application/models/Tracker/Sites/ReadMangaToday.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		$titleData = [];
+
+		$fullURL = $this->getFullTitleURL($title_url);
+
+		$content = $this->get_content($fullURL);
+
+		$data = $this->parseTitleDataDOM(
+			$content,
+			$title_url,
+			"//body/div[@class, 'content']/div/div/div/div/div/div/div/div/h1",
```
Comment (DakuTree) [inline application/models/Tracker/Sites/ReadMangaToday.php]: `[@class, 'content']` is invalid xpath, should be `[@class='content']`

File: application/models/Tracker/Sites/ReadMangaToday.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	public $chapterFormat = '/^[0-9\.]+$/';
+
+	public function getFullTitleURL(string $title_url) : string {
+		return "http://www.readmanga.today/{$title_url}";
+	}
+
+	public function getChapterData(string $title_url, string $chapter) : array {
+		//http://www.readmanga.today/saiki-kusuo-no-psi-nan/128
+		return [
+			'url'    => $this->getFullTitleURL($title_url).$chapter.'/',
```
Comment (DakuTree) [inline application/models/Tracker/Sites/ReadMangaToday.php]: This should be `$this->getFullTitleURL($title_url).'/'.$chapter`.

File: application/models/Tracker/Sites/ReadMangaToday.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			"//body/div[@class, 'content']/div/div/div/div/div/div/div/div/h1",
+			"//ul[contains(@class, 'chp_lst')]/li[1]/a[1]",
+			"span[@class='dte']",
+			"",
+			"404 Page Not Found"
+		);
+		if($data) {
+			$titleData['title'] = trim($data['nodes_title']->textContent);
+
+			$titleData['latest_chapter'] = preg_replace('/^.*\/([0-9\.]+)\/$/', '$1', (string) $data['nodes_chapter']->getAttribute('href'));
```
Comment (DakuTree) [inline application/models/Tracker/Sites/ReadMangaToday.php]: Regex here should be `/^.*\/([0-9\.]+)$/`, readmanga doesn't appear to use trailing slashes.

File: public/userscripts/manga-tracker.user.js [hunk 4]
```diff
@@ -1890,6 +1891,37 @@ let sites = {
 		}
 	}),
 
+
+  'www.readmanga.today' : extendSite({
+		setObjVars : function() {
+			this.segments      = window.location.pathname.replace(/^(.*\/)(?:[0-9]+\.html)?$/, '$1').split( '/' );
+
+			//FIXME: Is there a better way to do this? It just feels like an ugly way of setting vars.
+			this.page_count    = $('.list-switcher-2 > li > select[name=category_type').get(0).length;
```
Comment (DakuTree) [inline public/userscripts/manga-tracker.user.js]: Missing closing bracket here.

File: public/userscripts/manga-tracker.user.js [hunk 5]
```diff
@@ -1890,6 +1891,37 @@ let sites = {
...
+			this.title         = this.segments[1];
+			this.chapter       = this.segments[2];
+
+			this.title_url   = this.https + '://www.readmanga.today/'+this.title+'/';
+			this.chapter_url = this.title_url + this.chapter+'/';
+
+			//Might be easier to keep chapter_url different.
+			this.chapterListCurrent = this.chapter_url.slice(0,-1);
+			this.chapterList        = generateChapterList($('.jump-menu[name=chapter_list] > option:gt(0)'), 'value');
```
Comment (DakuTree) [inline public/userscripts/manga-tracker.user.js]: `.reverseObj()` needs to be used here so chapters are ordered from first to last.

File: application/models/Tracker/Sites/ReadMangaToday.php (hunk 1 above)
Comment (Vesnyx) [inline application/models/Tracker/Sites/ReadMangaToday.php]: Does that mean that it's also incorrect in MangaCow's?
It has 
"//ul[contains(@class, 'mng_chp')]/li[1]/a[1]"
https://github.com/DakuTree/manga-tracker/blob/master/application/models/Tracker/Sites/MangaCow.php#L29

Comment (DakuTree) [inline application/models/Tracker/Sites/ReadMangaToday.php]: `contains()` uses a different format than what is needed here.

Comment (Vesnyx) [inline application/models/Tracker/Sites/ReadMangaToday.php]: lol, okay. Totally missed the contains function there.

File: application/models/Tracker/Sites/ReadMangaToday.php (hunk 2 above)
Comment (Vesnyx) [inline application/models/Tracker/Sites/ReadMangaToday.php]: Once again, for comparison. URL scheme in mangacow seems similar, but uses this:
`'url'    => $this->getFullTitleURL($title_url).$chapter.'/',`

I assume this has something to do with a different chapter parameter.

Comment (DakuTree) [inline application/models/Tracker/Sites/ReadMangaToday.php]: MangaCow urls all have trailing slashes by default, while readmanga.today does not.

Comment (Vesnyx) [inline application/models/Tracker/Sites/ReadMangaToday.php]: Ah, that makes sense.

**Comment by DakuTree** [comment]
Site support should be live now.

**Pull request merged**